### ISSUE TITLE
Windows GitHub Desktop Commit Failed

### ISSUE THREAD

**Issue by dshepsis** [typicode/husky#250]
I'm unable to commit with a pre-commmit hook with Husky on GitHub Desktop on Windows. When I try to commit, I get a "Commit Failed" message and this file is produced:

[GitHubLog.txt](https://github.com/typicode/husky/files/1797796/GitHubLog.txt)

(I've modified it slightly to remove directory names.)

I am, however, able to commit just fine using the posh~git / PowerShell shell, using `git add .` followed by `git commit`. The pre-commit script works as expected here.

I've found a few other issues which seem to be related (namely #228, though that appears to have been fixed), but I'm not sure what is wrong here.

Any help is appreciated, but in the meanwhile, I can just use the Git shell instead of the desktop app.

**Comment by dshepsis** [comment]
I am using GitHub Desktop `Chocolate-Covered Yaks (3.3.4.0) 50415df` according to the "About" option. Maybe there's an issue with the difference between the versions you and I are running?

I don't recall exactly how I installed node, but I'm fairly certain it was through the nodejs.org website, like you suggested.

Running `node -v` in `cmd.exe` yields `v8.9.4`.

The issue still persists on my end.

**Comment by typicode** [comment]
Hmm not sure what the differences are between:
https://desktop.github.com/release-notes/ (most recent Feb 2018)
https://desktop.github.com/release-notes/windows/ (most recent Jan 2017)

Maybe it would be worth trying reinstalling GitHub desktop?

**Comment by noldors** [comment]
In ubuntu on version 0.15.0-rc.10 git commit has no effect too.
But on 0.15.0-rc.9 works fine.

**Comment by typicode** [comment]
@noldors rc.11 should fix that, thank you for the heads up!

**Comment by dshepsis** [comment]
I refactored the script I was using as a pre-commit script. Previously, it would directly alter one of the (potentially) staged files, which was causing issues with in GitHub desktop and posh Git. I've since learned that you're not meant to alter files during the pre-commit hook. The reason I gather for this is because it would result in alterations being committed which have never been reviewed by the committer, which could be quite dangerous.

The script now has an option for validating the existing output file against the expected output, rather than blindly overwriting the existing output file. If the output file doesn't match what's expected, the script gives an error message and exits with exit code 1 by setting `process.exitCode`.

I've also uninstalled the version of GitHub desktop I was using and reinstalled it from [here](https://desktop.github.com/). I can see now that the version I was using *was* old, though I have no idea why it never updated itself. In any case, that seems to have resolved the other half of the issue, which was the desktop client causing commits to fail even when no files were being changed by the pre-commit hook.

At this point, all of my issues have been resolved (at least as far as I can tell), so there this issue may be closed.
